### PR TITLE
Documents public API and types.

### LINKED ISSUE #67: Support godoc documentation
Ah and one thing: it would be nice to have godoc for most important Public APIs.

You can run godoc server locally 

```
go install -u golang.org/x/tools/cmd/godoc@latest
godoc -http=":6060"
```

Currently is totally empty for the project. This is how it does look like for an example package: https://pkg.go.dev/net/http

This is a standard way to check documentation by Go devs, so it's good to have at least some docs.

Here is a bit more about doc format: https://elliotchance.medium.com/godoc-tips-tricks-cda6571549b

_Originally posted by @roblaszczak in https://github.com/EventStore/EventStore-Client-Go/issues/65#issuecomment-928176517_

### PR DESCRIPTION
**Pull request by YoEight**
Fixes #67

### Breaking changes.
* Most enums got renamed to match Go naming conventions

### CODE REVIEW

File: esdb/types.go [hunk 1]
```diff
@@ -88,46 +112,58 @@ type Acl struct {
 	metaWriteRoles []string
 }
 
+// AddReadRoles Adds delete roles.
```
Comment (hayley-jean) [inline esdb/types.go:115]:
```suggestion
// AddReadRoles Adds read roles.
```

File: esdb/types.go [hunk 2]
```diff
@@ -137,26 +173,33 @@ type StreamMetadata struct {
...
 	m.maxCount = []uint64{value}
 }
 
+// SetMaxAge The maximum age of events allowed in the stream.
 func (m *StreamMetadata) SetMaxAge(value time.Duration) {
 	m.maxAge = []time.Duration{value}
 }
 
+// SetTruncateBefore The event number from which previous events can be scavenged. This is used to implement
+// soft-deletions of streams.
```
Comment (hayley-jean) [inline esdb/types.go:187]:
```suggestion
// deletions of streams.
```

File: esdb/types.go [hunk 3]
```diff
@@ -181,6 +226,8 @@ func (m *StreamMetadata) MaxAge() *time.Duration {
 	return &m.maxAge[0]
 }
 
+// TruncateBefore The event number from which previous events can be scavenged. This is used to implement soft-deletions
```
Comment (hayley-jean) [inline esdb/types.go:229]:
```suggestion
// TruncateBefore The event number from which previous events can be scavenged. This is used to implement deletions
```

File: esdb/types.go [hunk 4]
```diff
@@ -405,88 +480,87 @@ func StreamMetadataFromMap(props map[string]interface{}) (StreamMetadata, error)
...
+	// Type of filter.
+	Type FilterType
+	// String prefixes.
 	Prefixes []string
-	Regex    string
+	// Regex expression.
+	Regex string
 }
 
+// ExcludeSystemEventsFilter filters out event's with system's event types.
```
Comment (hayley-jean) [inline esdb/types.go:555]: Should this include the fact that system event types start with a `$`?
```suggestion
// ExcludeSystemEventsFilter filters out event's with system's event types, i.e. event types starting with `$`.
```

File: esdb/types.go [hunk 5]
```diff
@@ -508,32 +582,53 @@ type PersistentSubscriptionInfoHttpJson struct {
...
+// PersistentSubscriptionStats represents processing-related persistent subscription statistics.
 type PersistentSubscriptionStats struct {
-	AveragePerSecond              int64
-	TotalItems                    int64
-	CountSinceLastMeasurement     int64
+	// Average number of events per seconds.
+	AveragePerSecond int64
+	// Total number of events processed by subscriptions.
+	TotalItems int64
+	// Number of events seen since last measurement on this connection.
```
Comment (hayley-jean) [inline esdb/types.go:607]:
```suggestion
	// Number of events seen since last measurement on this subscription.
```

File: esdb/types.go [hunk 6]
```diff
@@ -508,32 +582,53 @@ type PersistentSubscriptionInfoHttpJson struct {
...
 }
 
+// PersistentSubscriptionStats represents processing-related persistent subscription statistics.
 type PersistentSubscriptionStats struct {
-	AveragePerSecond              int64
-	TotalItems                    int64
-	CountSinceLastMeasurement     int64
+	// Average number of events per seconds.
+	AveragePerSecond int64
+	// Total number of events processed by subscriptions.
```
Comment (hayley-jean) [inline esdb/types.go:605]:
```suggestion
	// Total number of events processed by this subscription.
```

**Review by hayley-jean — CHANGES_REQUESTED** [review]

File: esdb/client.go [hunk 7]
```diff
@@ -341,14 +357,16 @@ func (client *Client) SubscribeToStream(
...
 		}
 	}
 	defer cancel()
 	return nil, fmt.Errorf("failed to initiate subscription")
 }
 
-// SubscribeToAll ...
+// SubscribeToAll Subscribes to $all. This kind of subscription specifies a starting point (by default, the beginning
+// of a stream). This subscription will fetch all events until the end of the stream. Then, the subscription dispatches
+// subsequently written events.
```
[the remark this answers is not in the document]
Comment (YoEight) [reply esdb/client.go]: Done.

**Review by hayley-jean — DISMISSED** [review]

**Review by pvanbuijtene — APPROVED** [review]

**Review by pvanbuijtene — APPROVED** [review]

**Review by hayley-jean — APPROVED** [review]

**Pull request merged**